test/allow.js

```diff
@@ -14,117 +14,94 @@ verifiedSecretaryGraph.add(
   )
 const notVerifiedSecretaryGraph = rdf.graph()
 
-describe('allow() handler', () => {
-  describe('getUserId()', () => {
+describe('handlers/allow.js', () => {
+  describe('getRequestingWebId()', () => {
     const emptyFetchDocument = () => {}
-    it('no webId present', done => {
+
+    it('should return null if no webId and no onBehalfOf present', () => {
       let webId  // no secretary
-      let onBehalfOf = alice  // principal
-      getUserId(webId, onBehalfOf, emptyFetchDocument)
+      let onBehalfOf  // no principal
+
+      return getRequestingWebId(webId, onBehalfOf, emptyFetchDocument)
         .then(userId => {
-          assert.equal(userId, undefined,
-            'Should return undefined when no webId is present')
+          assert.equal(userId, null,
+            'Should return null when no webId is present')
+        })
+    })
+
+    it('should throw an error if onBehalfOf but no webId', done => {
+      let webId  // no secretary
+      let onBehalfOf = alice  // principal
+
+      getRequestingWebId(webId, onBehalfOf, emptyFetchDocument)
+        .catch(err => {
+          assert.equal(err.status, 400)
           done()
         })
     })
-    it('no onBehalfOf present', done => {
+
+    it('should return the webId if no delegation header present', () => {
       let webId = alice  // principal
       let onBehalfOf  // no delegation header
-      getUserId(webId, onBehalfOf, emptyFetchDocument)
+
+      return getRequestingWebId(webId, onBehalfOf, emptyFetchDocument)
         .then(userId => {
           assert.equal(webId, userId,
             'Should return webId when no onBehalfOf is present')
-          done()
-        })
-    })
-    it('throws 500 error if fetchDocument errors', done => {
```

Comment:
I removed that test because the fetch document function callback is no longer throwing a 500. I realized that the callback had no way of knowing the semantics of why it failed, whether for a 4xx-type reason or a 5xx-type. 
It'll be the responsibility of `fetchDocument()`, when we refactor it, to provide a more semantic error code.